src/material/table/testing/row-harness.ts

```diff
@@ -0,0 +1,85 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {ComponentHarness, HarnessPredicate} from '@angular/cdk/testing';
+import {RowHarnessFilters, CellHarnessFilters} from './table-harness-filters';
+import {MatCellHarness, MatHeaderCellHarness, MatFooterCellHarness} from './cell-harness';
+
+/** Harness for interacting with a standard Angular Material table row. */
+export class MatRowHarness extends ComponentHarness {
+  static hostSelector = '.mat-row';
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for a row with specific attributes.
+   */
+  static with(options: RowHarnessFilters = {}): HarnessPredicate<MatRowHarness> {
+    return new HarnessPredicate(MatRowHarness, options);
+  }
+
+  /** Gets all cells of the table row. */
+  async getCells(filter: CellHarnessFilters = {}): Promise<MatCellHarness[]> {
+    return this.locatorForAll(MatCellHarness.with(filter))();
+  }
+
+  /** Gets the text of the cells in the row. */
+  async getCellTextByIndex(filter: CellHarnessFilters = {}): Promise<string[]> {
```

Comment:
isn't there supposed to be another method like `getCellTextByColumnName`?

Reply:
Responded to it here https://github.com/angular/components/pull/17799#discussion_r358429036. I'll add it in a follow up since I'll have to move some things around and this PR has been pending for a while now.

Reply:
When you say "move some things around" does that mean breaking API changes? If so I'd rather have it in this PR. If you're just moving things under the hood and adding a new API a followup is fine

Reply:
It's just re-arranging some private functions, it won't be a breaking change.